--- clutter-gtk/gtk-clutter-actor.c	Thu Aug 21 21:53:43 2014
+++ clutter-gtk/gtk-clutter-actor.c	Wed Sep 24 12:05:20 2014
@@ -59,7 +59,7 @@
 #endif
 
 #ifdef CLUTTER_WINDOWING_WIN32
-#include <clutter/clutter-win32.h>
+#include <clutter/win32/clutter-win32.h>
 #endif
 #ifdef GDK_WINDOWING_WIN32
 #include <gdk/gdkwin32.h>
--- clutter-gtk/gtk-clutter-embed.c	Sun Sep 21 17:31:15 2014
+++ clutter-gtk/gtk-clutter-embed.c	Thu Dec 18 08:47:02 2014
@@ -578,7 +578,9 @@
                                  GtkAllocation *allocation)
 {
   GtkClutterEmbedPrivate *priv = GTK_CLUTTER_EMBED (widget)->priv;
+#if defined(GDK_WINDOWING_X11) && defined(CLUTTER_WINDOWING_X11)
   int scale_factor = gtk_widget_get_scale_factor (widget);
+#endif
 
   gtk_widget_set_allocation (widget, allocation);
 
--- clutter-gtk/Makefile.in	Mon Sep 22 17:45:46 2014
+++ clutter-gtk/Makefile.in	Wed Sep 24 12:04:29 2014
@@ -452,7 +452,7 @@
 libclutter_gtk_@CLUTTER_GTK_API_VERSION@_la_LIBADD = $(CLUTTER_GTK_DEPS_LIBS) $(LIBM)
 libclutter_gtk_@CLUTTER_GTK_API_VERSION@_la_LDFLAGS = \
 	$(CLUTTER_LT_LDFLAGS) \
-	-export-symbols-regex "^gtk_clutter.*"
+	-no-undefined -export-symbols-regex "^gtk_clutter.*"
 
 cluttergtkheadersdir = $(includedir)/clutter-gtk-@CLUTTER_GTK_API_VERSION@/clutter-gtk
 
--- clutter-gtk/gtk-clutter-util.c	Thu Aug 28 21:44:44 2014
+++ clutter-gtk/gtk-clutter-util.c	Mon Aug 17 21:21:00 2015
@@ -95,7 +95,9 @@
       GDK_IS_WIN32_DISPLAY (display))
     {
       /* let GTK+ be in charge of the event handling */
+      /* PATCH: gtk event handling broken at the moment
       clutter_win32_disable_event_retrieval ();
+      */
     }
   else
 #endif
