--- clutter-gtk/gtk-clutter-embed.c	Sun Sep 21 17:31:15 2014
+++ clutter-gtk/gtk-clutter-embed.c	Thu Dec 18 08:47:02 2014
@@ -578,7 +578,9 @@
                                  GtkAllocation *allocation)
 {
   GtkClutterEmbedPrivate *priv = GTK_CLUTTER_EMBED (widget)->priv;
+#if defined(GDK_WINDOWING_X11) && defined(CLUTTER_WINDOWING_X11)
   int scale_factor = gtk_widget_get_scale_factor (widget);
+#endif
 
   gtk_widget_set_allocation (widget, allocation);
 
--- clutter-gtk/gtk-clutter-util.c	Thu Aug 28 21:44:44 2014
+++ clutter-gtk/gtk-clutter-util.c	Mon Aug 17 21:21:00 2015
@@ -95,7 +95,9 @@
       GDK_IS_WIN32_DISPLAY (display))
     {
       /* let GTK+ be in charge of the event handling */
+      /* PATCH: gtk event handling broken at the moment
       clutter_win32_disable_event_retrieval ();
+      */
     }
   else
 #endif
