--- ext/dvdread/dvdreadsrc.c	Mon Jun 01 01:36:58 2015
+++ ext/dvdread/dvdreadsrc.c	Sat Oct 03 07:05:33 2015
@@ -1791,7 +1791,14 @@
 #ifdef ENABLE_NLS
   GST_DEBUG ("binding text domain %s to locale dir %s", GETTEXT_PACKAGE,
       LOCALEDIR);
+#ifdef G_OS_WIN32
+    gchar *_glib_get_locale_dir (void);
+    gchar *localedir = _glib_get_locale_dir ();
+    bindtextdomain (GETTEXT_PACKAGE, localedir);
+    g_free (localedir);
+#else
   bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+#endif
   bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 #endif /* ENABLE_NLS */
 
--- gst/asfdemux/gstasf.c	Sun Oct 19 14:08:33 2014
+++ gst/asfdemux/gstasf.c	Sat Oct 03 07:05:33 2015
@@ -39,7 +39,14 @@
 #ifdef ENABLE_NLS
   GST_DEBUG ("binding text domain %s to locale dir %s", GETTEXT_PACKAGE,
       LOCALEDIR);
+#ifdef G_OS_WIN32
+    gchar *_glib_get_locale_dir (void);
+    gchar *localedir = _glib_get_locale_dir ();
+    bindtextdomain (GETTEXT_PACKAGE, localedir);
+    g_free (localedir);
+#else
   bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+#endif
   bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 #endif /* ENABLE_NLS */
 
--- ext/twolame/gsttwolamemp2enc.c	Fri Sep 11 15:22:09 2015
+++ ext/twolame/gsttwolamemp2enc.c	Sat Oct 03 07:05:33 2015
@@ -875,7 +875,14 @@
 #ifdef ENABLE_NLS
   GST_DEBUG ("binding text domain %s to locale dir %s", GETTEXT_PACKAGE,
       LOCALEDIR);
+#ifdef G_OS_WIN32
+    gchar *_glib_get_locale_dir (void);
+    gchar *localedir = _glib_get_locale_dir ();
+    bindtextdomain (GETTEXT_PACKAGE, localedir);
+    g_free (localedir);
+#else
   bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+#endif
   bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 #endif /* ENABLE_NLS */
 
--- ext/lame/plugin.c	Sun Oct 19 14:08:33 2014
+++ ext/lame/plugin.c	Sat Oct 03 07:05:33 2015
@@ -30,7 +30,14 @@
 plugin_init (GstPlugin * plugin)
 {
 #ifdef ENABLE_NLS
+#ifdef G_OS_WIN32
+    gchar *_glib_get_locale_dir (void);
+    gchar *localedir = _glib_get_locale_dir ();
+    bindtextdomain (GETTEXT_PACKAGE, localedir);
+    g_free (localedir);
+#else
   bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+#endif
   bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 #endif /* ENABLE_NLS */
 
