--- gtk/gtkiconcache.c	Thu Apr 09 17:44:23 2015
+++ gtk/gtkiconcache.c	Sun Apr 12 15:37:18 2015
@@ -106,13 +106,14 @@
     goto done;
 
 #ifdef G_OS_WIN32
-
+#ifndef __MINGW32__
 /* Bug 660730: _fstat32 is only defined in msvcrt80.dll+/VS 2005+ */
 /*             or possibly in the msvcrt.dll linked to by the Windows DDK */
 /*             (will need to check on the Windows DDK part later) */
 #if (_MSC_VER >= 1400 || __MSVCRT_VERSION__ >= 0x0800)
 #undef fstat /* Just in case */
 #define fstat _fstat32  
+#endif
 #endif
 #endif
 
--- gtk/org.gtk.Settings.FileChooser.gschema.xml	Tue Mar 03 02:34:03 2015
+++ gtk/org.gtk.Settings.FileChooser.gschema.xml	Sun Apr 12 15:39:07 2015
@@ -64,7 +64,7 @@
       </description>
     </key>
     <key type="b" name="sort-directories-first">
-      <default>false</default>
+      <default>true</default>
       <summary>Show folders first</summary>
       <description>
         If set to true, then folders are shown before files in the list.
--- gtk/updateiconcache.c	Thu Apr 09 17:44:23 2015
+++ gtk/updateiconcache.c	Sun Apr 26 23:22:41 2015
@@ -605,7 +605,7 @@
   gboolean dir_added = FALSE;
   guint dir_index = 0xffff;
   
-  dir_path = g_build_path ("/", base_path, subdir, NULL);
+  dir_path = g_build_path (G_DIR_SEPARATOR_S, base_path, subdir, NULL);
 
   /* FIXME: Use the gerror */
   dir = g_dir_open (dir_path, 0, NULL);
@@ -631,7 +631,7 @@
 	  gchar *subsubdir;
 
 	  if (subdir)
-	    subsubdir = g_build_path ("/", subdir, name, NULL);
+	    subsubdir = g_build_path (G_DIR_SEPARATOR_S, subdir, name, NULL);
 	  else
 	    subsubdir = g_strdup (name);
 	  directories = scan_directory (base_path, subsubdir, files, 
--- Makefile.in	Thu Apr 09 17:49:53 2015
+++ Makefile.in	Sun Apr 12 15:37:18 2015
@@ -550,7 +550,7 @@
 	|| { echo "Gtk+Tests:ERROR: Failed to start Xvfb environment for X11 target tests."; exit 1; } \
 	&& DISPLAY=:$$XID && export DISPLAY
 
-SRC_SUBDIRS = util gdk gtk libgail-util modules demos tests testsuite examples
+SRC_SUBDIRS = util gdk gtk libgail-util modules demos
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 MAINTAINERCLEANFILES = \
